gradio/utils.py

```diff
@@ -689,6 +690,39 @@ def is_update(val):
     return type(val) is dict and "update" in val.get("__type__", "")
 
 
+def get_continuous_fn(fn, every):
+    def continuous_fn(*args):
+        called_before = False
+        while True:
+            output = fn(*args)
+            if called_before:
+                time.sleep(every)
+            else:
+                called_before = True
+            yield output
+
+    return continuous_fn
```

Comment:
```suggestion
def get_continuous_fn(fn, every):
    def continuous_fn(*args):
        while True:
            output = fn(*args)
            yield output
            time.sleep(every)
    return continuous_fn
```

Also, would it be better to use `asyncio.sleep` here instead of `time.sleep` so that the queue worker is free to do something else in the meantime?

Reply:
Ah I see so you later avoid putting such events in the queue altogether because they don't get completed. So I'm not sure how much of a difference it makes to use `asyncio.sleep` but I do think it's probably better than having a background thread just sit there doing nothing

Reply:
Lol love this refactor! Thank you.

Yea originally I had this as an async function but the problem with that is that you only want to sleep after the first function call. Tricky to implement without a generator and we don't currently support async generators. Will think about it though (may add async generators to #2552 )